functions/New-DbaDbUser.ps1

```diff
@@ -0,0 +1,247 @@
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [Alias("Databases")]
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [switch]$IncludeSystem,
+        [Alias("Name", "LoginName")]
+        [parameter(ParameterSetName = "Login")]
+        [string[]]$Login,
+        # [parameter(ParameterSetName = "Contained")]
+
+        [parameter(ParameterSetName = "Login")]
+        [parameter(ParameterSetName = "NoLogin")]
+        [string[]]$Username,
+        # [parameter(ParameterSetName = "Contained")]
+
+        # [Security.SecureString]$Password,
+
+        [switch]$Force,
+        [switch][Alias('Silent')]
+        $EnableException
+    )
+    
+    begin {
+        
+        
+    }
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                Write-Message -Level Verbose -Message "Connecting to $instance"
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+                return
+            }
+            
+            $databases = $server.Databases | Where-Object Status -EQ "normal"
+            
+            if ($Database) {
+                $databases = $databases | Where-Object Name -In $Database
+            }
+            if ($ExcludeDatabase) {
+                $databases = $databases | Where-Object Name -NotIn $ExcludeDatabase
+            }
+            If (-Not $IncludeSystem) {
+                $databases = $databases | Where-Object IsSystemObject -NE $true
+            }
+            
+            
+            foreach ($db in $databases) {
+                Write-Message -Level Verbose -Message "Add users to Database $db on target $server";
+                
+                switch ($PSCmdlet.ParameterSetName) {
+                    "Login" {
+                        # Creates a user with Login
+                        Write-Message -Level Verbose -Message "Using UserType: SqlLogin"
+                        
+                        if ($PSBoundParameters.Keys -notcontains 'Login') {
+                            Stop-Function -Message "Parameter -Login is required " -Target $instance
+                            return
+                        }
+                        
+                        
+                        if ($Login.GetType().Name -eq 'Login') {
+                            $smoLogin = $Login
+                        }
+                        else {
+                            #get the login associated with the given name.
+                            $smoLogin = $server.Logins | Where-Object Name -eq $Login
+                            if ($smoLogin -eq $null) { Stop-Function -Message "Invalid Login: $Login is not found on $Server" -Target $instance; return }
```

Comment:
Format, this needs to be split across multiple lines for readability.

The return needs to be removed and `Stop-Function` should have the `-Continue` parameter added. By using the return you stop the call of the function completely. You provide the ability to pass in multiple logins, so the first one that is now found stops any of the other ones from being processed.

Which if multiple logins are passed in `Where-Object Name -eq $Login` will always come up empty. Without testing you likely need to provide a loop to iterate over each login passed in.